src/main/java/ua/hillel/treshchun/lessons/lesson13Homework14/StringClass.java

```diff
@@ -0,0 +1,73 @@
+package ua.hillel.treshchun.lessons.lesson13Homework14;
+
+import java.util.Scanner;
+
+public class StringClass {
+
+    public String readWord() {
+        Scanner scanner = new Scanner(System.in);
+        String scannnedString;
+        while (!scanner.hasNextLine()) {
+            System.out.println("Please enter a valid string");
+        }
+        scannnedString = scanner.nextLine();
+        while (scannnedString.isEmpty() || scannnedString.isBlank()) {
+            System.out.println("Please enter a valid string: ");
+            scannnedString = scanner.nextLine();
+        }
+        return scannnedString;
+    }
+
+    public int findSymbolOccurance(String str, char chr) {
+        return (int) str.chars().filter(c -> c == chr).count();
+    }
+
+    public int findWordPosition(String source, String target) {
+        return source.lastIndexOf(target);
+    }
+
+    public String stringReverse(String inputStr) {
+        String result = "";
+        for (int i = inputStr.length() - 1; i > -1; i--) {
+            result += inputStr.charAt(i);
```

Comment:
StringBuilder.reverse(); could be used but your solution also works
remember what I told about str+="str" concatination especially in loops. For huge input strings with length more than 100k symbols this method could lead to OutOfMemoryError. It should be fixed, replaced by any StringBuilder usage. -2 score points till your fix